tests/disaster-recovery/sc_arbiter/test_shutdowns.py

```diff
@@ -0,0 +1,487 @@
+        setup_logwriter_cephfs_workload_factory,
+        setup_logwriter_rbd_workload_factory,
+        logreader_workload_factory,
+    ):
+        """
+        This test will test the shutdown scenarios when active-active CephFS and RBD workloads
+        is running.
+        Steps:
+            1) Run both the logwriter and logreader CephFS and RBD workloads
+               CephFS workload uses RWX volume and RBD workload uses RWO volumes
+            2) Reset the connection scores for the mons
+            3) Induce the shutdown
+               In case of normal shutdown we shut-down and wait for about 15 mins
+               before start of nodes whereas immediate shutdown would involve starting
+               nodes immediately just after 5 mins
+            4) Make sure ceph is accessible during the crash duration
+            5) Repeat the shutdown process as many times as number of iterations
+            6) Make sure logreader job pods have Completed state.
+               Check if there is any write or read pause. Fail only when neccessary.
+            7) Delete the old logreader job and create new logreader job to verify the data corruption
+            8) Make sure there is no data loss
+            9) Validate the connection scores
+            10) Do a complete cluster sanity and make sure there is no issue post recovery
+
+        """
+        sc_obj = StretchCluster()
+
+        if immediate:
+            sc_obj.default_shutdown_durarion = 5
+
+        # Run the logwriter cephFs workloads
+        (
+            sc_obj.cephfs_logwriter_dep,
+            sc_obj.cephfs_logreader_job,
+        ) = setup_logwriter_cephfs_workload_factory(read_duration=0)
+
+        # Generate 5 minutes worth of logs before inducing the netsplit
+        log.info("Generating 5 mins worth of log")
+        time.sleep(10)
+
+        sc_obj.get_logwriter_reader_pods(label=constants.LOGWRITER_CEPHFS_LABEL)
+        sc_obj.get_logwriter_reader_pods(label=constants.LOGREADER_CEPHFS_LABEL)
+        sc_obj.get_logwriter_reader_pods(
+            label=constants.LOGWRITER_RBD_LABEL, exp_num_replicas=2
+        )
+
+        zones = constants.ZONES_LABELS
+        zones.remove("arbiter")
+
+        start_time = datetime.now(timezone.utc)
+        for i in range(iteration):
+            log.info(f"------ Iteration {i+1} ------")
+            if not immediate:
+                start_time = datetime.now(timezone.utc)
+
+            # note the file names created
+            sc_obj.get_logfile_map(label=constants.LOGWRITER_CEPHFS_LABEL)
+            sc_obj.get_logfile_map(label=constants.LOGWRITER_RBD_LABEL)
+
+            # Fetch the nodes in zone that needs to be crashed
+            zone = random.choice(zones)
+            nodes_to_shutdown = sc_obj.get_nodes_in_zone(zone)
+
+            assert (
+                len(nodes_to_shutdown) != 0
+            ), f"There are 0 zone nodes labeled as topology.kubernetes.io/zone={zone}!!"
+
+            try:
```

Comment:
wait_for_nodes_status already throws the exception when the node doesnt reach the desired state, do we still need to handle in try and except

Reply:
done